### PR TITLE
Backend

### PR DESCRIPTION
**Pull request by Athinanarof**
## What is the problem/goal being addressed?
Translate some functions needed to create a random test case.

## What is the solution to this problem?
While creating the test case suggested for the random function, I worked on some related functions: range, map, and bool.and.
I made small corrections to Bool.fs and formatted it; test cases are now running.
I translated the range and included test cases.
I added List.map test cases.
I made random_v0 inclusive of a.

## What are the changes here? How do they solve the problem, and what other product impacts do they cause?
*Please include before/after screenshots/gifs if appropriate*

## How are you sure this works/how was this tested?
Test cases are running and returning the correct results.

## What is the reversion plan if this fails after shipping?
Updates can always be reverted.

### CODE REVIEW

File: fsharp-backend/tests/testfiles/int.tests [hunk 1]
```diff
@@ -36,4 +36,5 @@ Int.subtract_v0 1 0 = 1
 Int.subtract_v0 -55 -55 = 0
 Int.multiply_v0 8 8 = 64
 Int.multiply_v0 5145 5145 = 26471025
-
+List.map_v0 (List.map_v0 (List.range_v0 1 5) (fun x -> Int.random_v0 1 2)) (fun x -> Bool.and_v0 (Int.greaterThanOrEqualTo_v0 x 0) (Int.lessThanOrEqualTo_v0 x 1)) = [true; true; true; true; true]
```
Comment (pbiggar) [inline fsharp-backend/tests/testfiles/int.tests:39]:
```suggestion
List.map_v0 (List.map_v0 (List.range_v0 1 100) (fun x -> Int.random_v0 1 2)) (fun x -> Bool.and_v0 (Int.greaterThanOrEqualTo_v0 x 0) (Int.lessThanOrEqualTo_v0 x 1)) = [true; true; true; true; true]
```

**Review by pbiggar — CHANGES_REQUESTED** [review]
Looks great. There's a handful of changes to make, but this is coming along nicely! 😊

**Comment by pbiggar** [comment]
Note there are merge conflicts too!

File: fsharp-backend/src/LibExecution/StdLib/LibInt.fs [hunk 2]
```diff
@@ -328,7 +328,7 @@ let fns : List<BuiltInFn> =
         (function
         (*( +1 as Random.int is exclusive *)
         | _, [ DInt a; DInt b ] ->
-            Value(DInt(a + one + (Runtime.random.Next((b - a) |> int) |> bigint)))
+            Value(DInt(a + (Runtime.random.Next((b - a) |> int) |> bigint)))
```
[the remark this answers is not in the document]
Comment (Athinanarof) [reply fsharp-backend/src/LibExecution/StdLib/LibInt.fs:331]: Yes, that was happening with random_v0 and v1. I have removed both comments and adjusted v1.

**Pull request merged**